### PR TITLE
Added `ScrollBox`

### PR DESCRIPTION
**Pull request by MrGVSV**
This adds a `ScrollBox` widget that can be used to scroll overflowing content.

Additionally, it adds a `ScrollEvent` type, among others.

## Notable Issues

There are a few issues with this solution:

* It re-renders on every scroll (not sure if that's even fixable)
* It requires a few renders to calculate all the layouts which can cause a few frames of flickering
* Padding that accounts for the scrollbars works differently if the content is manually sized

## Screenshots

https://user-images.githubusercontent.com/49806985/154824645-9ff098b0-d764-4309-8908-d275d40431e5.mp4

### CODE REVIEW

File: src/widgets/scroll/scroll_bar.rs
Comment (StarArawn) [inline src/widgets/scroll/scroll_bar.rs:105]: Should be a regular comment.

File: src/widgets/scroll/scroll_bar.rs
Comment (MrGVSV) [reply src/widgets/scroll/scroll_bar.rs:68]: Yeah the goal was to make it so we could calculate the content layout in `ScrollContent` and update the size info on `ScrollContext`. I think we need a better way of getting that info (maybe with a `LayoutEvent` event type?).

It was also supposed to allow nested `ScrollBox` widgets, but I just tested it and it seems to not work like I thought it would :(

In fact, if I wrap my content in an `Element` it just flat out doesn't work. I think maybe it sizes to fit the parent rather than its children? 🤔

Comment (MrGVSV) [reply src/widgets/scroll/scroll_bar.rs:68]: I think #86 will fix this issue

**Comment by MrGVSV** [comment]
Marking as a draft until #95 is merged so that some of these commits can be rebased.

However, it's mostly working now. Still not perfect, but I'm not sure we want to keep holding off _some_ type of scrolling capability— even if it has a few... _quirks_.

**Review by StarArawn — APPROVED** [review]

**Pull request merged**
